packages/dds/tree/src/simple-tree/api/schemaFactoryRecursive.ts

```diff
@@ -159,37 +153,101 @@ export function createFieldSchemaUnsafe<
+	// These match whats defined in the recursive methods on `SchemaFactory` except they do not use `Unenforced`.
+	{
+		[NodeKind.Object]: T["info"] extends RestrictiveStringRecord<ImplicitFieldSchema>
+			? InsertableObjectFromSchemaRecord<T["info"]>
+			: unknown;
+		[NodeKind.Array]: T["info"] extends ImplicitAllowedTypes
+			? Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>>
+			: unknown;
+		[NodeKind.Map]: T["info"] extends ImplicitAllowedTypes
+			? Iterable<[string, InsertableTreeNodeFromImplicitAllowedTypes<T["info"]>]>
+			: unknown;
+		[NodeKind.Leaf]: unknown;
+	}[T["kind"]],
+	// ImplicitlyConstructable: recursive types are currently not implicitly constructable.
+	false,
+	// Info: What's passed to the method to create the schema. Constraining these here should be about as effective as if the actual constraints existed on the actual method itself.
+	{
+		[NodeKind.Object]: RestrictiveStringRecord<ImplicitFieldSchema>;
+		[NodeKind.Array]: ImplicitAllowedTypes;
+		[NodeKind.Map]: ImplicitAllowedTypes;
+		[NodeKind.Leaf]: unknown;
+	}[T["kind"]]
+>;
+
+/**
+ * Workaround for "Type instantiation is excessively deep and possibly infinite.ts" errors.
+ * @remarks
+ *
+ * Generally this workaround should be avoided if possible,
+ * especially for exported types as it is likely to result in issues when exporting or importing schema forcing the user to use the workaround as well.
+ * This is particularly problematic since in some cases the user replicating the pattern might not be possible.
+ * Additionally which cases hit these limits may vary based on TypeScript version and implementation details of this library.
+ *
+ * This workaround is provided and documented as a last resort to be able to keep an app compiling.
+ * Future version of SharedTree should provide schema type erasure functionality as a better alternative for most cases.
+ *
+ * When TypeScript gives the error "Error (TS2589) Type instantiation is excessively deep and possibly infinite." on the invocation of `ValidateRecursiveSchema`
+ * for a large schema, it can sometimes be worked around by repeating the usage of the type multiple times.
+ * This works because the TypeScript compiler caches some of the intermediate results from the first usage, and thus can get further on the second.
+ *
+ * This utility can be referenced when applying this pattern.
+ * For recursive types this can be used directly:
+ *
+ * ```typescript
+ * {
+ *     // @ts-expect-error Recursion limit
+ *     type _check1 = FixRecursiveRecursionLimit<typeof LargeType>;
+ *     type _check2 = FixRecursiveRecursionLimit<typeof LargeType>;
+ *     type _check3 = ValidateRecursiveSchemaTemplate<typeof LargeType>;
+ * }
+ * ```
+ *
+ * For non-recursive types, they can be ported to the more flexible recursive APIs and use the pattern above.
+ *
+ * Non-recursive types can also use this workaround by making a duplicate copy of the schema causing the error but written using the recursive APIs.
```

Comment:
Nit (for readability)
```suggestion
 * Non-recursive types can also use this workaround by making a duplicate copy of the problematic schema, but written using the recursive APIs.
```